ms2pip/ms2pipC.py

```diff
@@ -790,66 +654,27 @@ def peakcount(x):
     return c / len(x)
 
 
-def argument_parser():
-    parser = argparse.ArgumentParser()
-    parser.add_argument("pep_file", metavar="<PEPREC file>", help="list of peptides")
-    parser.add_argument(
-        "-c",
-        metavar="CONFIG_FILE",
-        action="store",
-        required=True,
-        dest="config_file",
-        help="config file",
-    )
-    parser.add_argument(
-        "-s",
-        metavar="MGF_FILE",
-        action="store",
-        dest="spec_file",
-        help=".mgf MS2 spectrum file (optional)",
-    )
-    parser.add_argument(
-        "-w",
-        metavar="FEATURE_VECTOR_OUTPUT",
-        action="store",
-        dest="vector_file",
-        help="write feature vectors to FILE.{pkl,h5} (optional)",
-    )
-    parser.add_argument(
-        "-x",
-        action="store_true",
-        default=False,
-        dest="correlations",
-        help="calculate correlations (if MGF is given)",
-    )
-    parser.add_argument(
-        "-t",
-        action="store_true",
-        default=False,
-        dest="tableau",
-        help="create Tableau Reader file",
-    )
-    parser.add_argument(
-        "-m",
-        metavar="NUM_CPU",
-        action="store",
-        dest="num_cpu",
-        help="number of CPUs to use (default: all available)",
-    )
-    args = parser.parse_args()
-
-    if not args.num_cpu:
-        args.num_cpu = multiprocessing.cpu_count()
-
-    return (
-        args.pep_file,
-        args.spec_file,
-        args.vector_file,
-        args.config_file,
-        int(args.num_cpu),
-        args.correlations,
-        args.tableau,
-    )
+def get_intense_mzs(mzs, intensity, n=3):
+    return [x[0] for x in sorted(zip(mzs, intensity), key=itemgetter(1), reverse=True)[:n]]
+
+
+def match_mzs(mzs, predicted, max_error=0.02):
+    current = 0
+    for pred in predicted:
+        current = bisect.bisect_right(mzs, pred - max_error, lo=current)
+        if current >= len(mzs) or mzs[current] > pred + max_error:
+            return False
+    return current < len(mzs)
+
+
+def get_predicted_peaks(results):
+    mz_bufs, prediction_bufs, _, _, _, pepid_bufs = zip(*(r.get() for r in results))
+
+    return dict(zip(itertools.chain.from_iterable(pepid_bufs),
+                    (sorted(get_intense_mzs(np.concatenate(mzs[0], axis=None),
+                                            np.concatenate(mzs[1], axis=None)))
+                     for mzs in zip(itertools.chain.from_iterable(mz_bufs),
+                                    itertools.chain.from_iterable(prediction_bufs)))))
```

Comment:
Maybe we can move the match_spectra functionality to a separate module in the future?